### PR TITLE
kubectl: Allow parallelizing diffing

### PR DESCRIPTION
**Pull request by brancz**
#### What type of PR is this?

/kind feature
/sig cli

#### What this PR does / why we need it:

We noticed that `kubectl diff` runs take a very long time in our CI/CD pipelines, so we investigated and figured out that diffing non concurrently so the more objects are being diffed the longer it takes. To combat this at least a bit and give the user some control, this PR adds the ability to specify the concurrency with which the diff is executed.

In our experiments, this resulted in exactly what we were looking for, roughly halving the execution time every time we double the concurrency.

We actually did this on a live stream, so if the thought process and profiling data we used to support this is interesting, here is the recording: https://www.youtube.com/watch?v=p17uA4ymdbs

#### Which issue(s) this PR fixes:

The implementation is so small I decided to just open the PR.

#### Does this PR introduce a user-facing change?

```release-note
Add `--concurrency` flag to configure the concurrency of `kubectl diff` execution, defaults to 1
```

We'll probably want to iterate on the flag name a little, but I just went with something for now, of course I'm very happy to rename this to anything else.

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go [hunk 1]
```diff
@@ -166,6 +167,7 @@ func NewCmdDiff(f cmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Co
 	cmd.Flags().StringArray("prune-allowlist", []string{}, "Overwrite the default whitelist with <group/version/kind> for --prune")
 	cmd.Flags().Bool("prune", false, "Include resources that would be deleted by pruning. Can be used with -l and default shows all resources would be pruned")
 	cmd.Flags().BoolVar(&options.ShowManagedFields, "show-managed-fields", options.ShowManagedFields, "If true, include managed fields in the diff.")
+	cmd.Flags().IntVar(&options.DiffConcurrency, "diff-concurrency", 1, "Number of objects to process in parallel when diffing against the live version. Larger number = faster, but more memory, I/O and CPU over that shorter period of time.")
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go:170]: Nit: can we make the flag just `concurrency`?

Comment (brancz) [reply staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go:170]: Of course! On it!

Comment (brancz) [reply staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go]: Done.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**